app/screens/send-bitcoin-screen/send-bitcoin-success-screen.tsx

```diff
@@ -11,37 +11,95 @@ import { RootStackParamList } from "@app/navigation/stack-param-lists"
 import { useNavigation } from "@react-navigation/native"
 import { StackNavigationProp } from "@react-navigation/stack"
 import { makeStyles, Text } from "@rneui/themed"
-import { View } from "react-native"
+import { View, Alert } from "react-native"
 import { testProps } from "../../utils/testProps"
-
-const useStyles = makeStyles(() => ({
-  contentContainer: {
-    flexGrow: 1,
-  },
-  successText: {
-    fontSize: 18,
-    textAlign: "center",
-    marginTop: 20,
-  },
-  Container: {
-    flex: 1,
-    justifyContent: "center",
-    alignItems: "center",
-  },
-}))
+import Rate from "react-native-rate"
+import { ratingOptions } from "@app/config"
+import crashlytics from "@react-native-firebase/crashlytics"
+import { useApolloClient } from "@apollo/client"
+import { useFeedbackModalShownQuery } from "@app/graphql/generated"
+import { setFeedbackModalShown } from "@app/graphql/client-only-query"
+import { SuggestionModal } from "./suggestion-modal"
+import { logAppFeedback } from "@app/utils/analytics"
 
 const SendBitcoinSuccessScreen = () => {
   const styles = useStyles()
+  const [isFeedbackModalActive, setIsFeedbackModalActive] = React.useState(false)
 
+  const [showImprovement, setShowImprovement] = React.useState(false)
   const navigation =
     useNavigation<StackNavigationProp<RootStackParamList, "sendBitcoinSuccess">>()
 
+  const client = useApolloClient()
+  const feedbackShownData = useFeedbackModalShownQuery()
+  const modalShown = feedbackShownData?.data?.feedbackModalShown
+
+  const dismiss = () => {
+    logAppFeedback({
+      isEnjoingApp: false,
+    })
+    setIsFeedbackModalActive(false)
+    setShowImprovement(true)
+  }
+
+  const rateUs = () => {
+    logAppFeedback({
+      isEnjoingApp: true,
+    })
+    Rate.rate(ratingOptions, (success, errorMessage) => {
+      if (success) {
+        crashlytics().log("User went to the review page")
+      }
+      if (errorMessage) {
+        crashlytics().recordError(new Error(errorMessage))
+      }
+    })
+  }
+
   const { LL } = useI18nContext()
+  const FEEDBACK_DELAY = 3000
   const CALLBACK_DELAY = 3000
   useEffect(() => {
+    if (!modalShown) {
+      const feedbackTimeout = setTimeout(
+        () => setIsFeedbackModalActive(true),
+        FEEDBACK_DELAY,
+      )
+      return () => {
+        clearTimeout(feedbackTimeout)
+        setFeedbackModalShown(client, true)
```

Comment:
`setFeedbackModalShown` should be set at the same time as the `Alert.alert` is called in line 65. Currently it is part of the hook clean up, so there is the potential that it does not get shown because the hook gets cleaned up before the modal is shown.

Reply:
This is still not resolved

Reply:
I see. I think it would be good to add back in the alertShown then.